### PR TITLE
Update to Gradle 8 with higher JDK Support

### PR DESCRIPTION
**Pull request by xeruf**
supports higher JDK versions, fancy dokka documentation and a lot more improvements including folder structure

### CODE REVIEW

File: server/src/test/java/sc/server/network/LobbyGameTest.kt [hunk 1]
```diff
@@ -77,14 +64,13 @@ class LobbyGameTest: WordSpec({
...
                     admin.control(room.id).step(true)
                     val result = roomListener.waitForMessage(GameResult::class)
                     playerHandlers[0].gameResult shouldBe result
-                    result.win shouldBe Team.ONE
-                    result.isRegular shouldBe false
-                    result.win?.reason.shouldBeInstanceOf<Violation.LEFT>()
+                    result.win?.winner shouldBe Team.TWO
+                    result.isRegular shouldBe true
                     admin.closed shouldBe false
                 }
```
Comment (soerendomroes) [inline server/src/test/java/sc/server/network/LobbyGameTest.kt]: Der Test scheint sich sehr geändert zu haben. Hat sich hier auch das Verhalten geändert?

**Review by soerendomroes — COMMENTED** [review]
Ich verstehe nicht ganz, warum jetzt alle plugins verschoben werden.

Hattest du getestet, ob man mit dem SimpleClient für die Schüler auch mit einer niedrigeren Java Version auskommt?

**Comment by xeruf** [comment]
> Ich verstehe nicht ganz, warum jetzt alle plugins verschoben werden.

damit es für die zukunft leichter zu maintainen bleibt wenn wir rotieren

> Hattest du getestet, ob man mit dem SimpleClient für die Schüler auch mit einer niedrigeren Java Version auskommt?

JDK 17 ist das minimum mit Gradle 9, dann müssten die Schüler die gradle version downgraden oder das ohne gradle machen

> Der build failed bei mir noch:

vorher mal `./gradlew clean` ausgeführt?

**Comment by soerendomroes** [comment]
@xeruf 

Leider baut das bei mir (auch nach clean) nicht:

```
* What went wrong:
Some problems were found with the configuration of task ':server:startScripts' (type 'CreateStartScripts').
  - Gradle detected a problem with the following location: '/home/sdo/Documents/socha/Development/gui/backend/server/build/runnable/server.jar'.
    
    Reason: Task ':server:startScripts' uses this output of task ':server:createStartScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
```

Klappt das bei dir einfach?

**Review by soerendomroes — DISMISSED** [review]

**Comment by soerendomroes** [comment]
https://en.wikipedia.org/wiki/Java_version_history

Java 24 ist nicht das neueste Java mit einer kleineren Nummer als 25. Die LTS Versionen werden weiter geupdated, Java 24 ist nicht mehr supported. Es lohnt sich entsprechend beim Test sowas wie 8, 11, 17 und 21 zu testen.

Und wir hatten uns auf Gradle 8.14.X geeinigt, da dies auch noch Java 8 supported.
https://docs.gradle.org/current/userguide/compatibility.html

File: .gitignore [hunk 2]
```diff
@@ -11,4 +11,5 @@ out/
 # Artifacts from test runs
 logs/
 log/
-replays/
\ No newline at end of file
+replays/
+plans/
```
[the remark this answers is not in the document]
Comment (xeruf) [reply .gitignore]: das ist für agentic coding

**Comment by xeruf** [comment]
wir nutzen shadow 9.1 weswegen das backend bauen mindestens jdk 11 braucht, aber teilnehmer können trotzdem noch jdk 8 nutzen weil ich shadow da auf 8.3 gesetzt hab - das ist kein problem
https://github.com/GradleUp/shadow?tab=readme-ov-file#compatibility-matrix

File: server/src/test/java/sc/server/network/LobbyGameTest.kt (hunk 1 above)
Comment (xeruf) [reply server/src/test/java/sc/server/network/LobbyGameTest.kt:75]: ich schaus mir grad an - scheint ein testspezifischer edge case zu sein - ein raum mit nur einem spieler wird gestartet

Comment (xeruf) [reply server/src/test/java/sc/server/network/LobbyGameTest.kt:75]: puh, das war eine Reise das zu debuggen - aber jetzt ist es wieder zurück korrigiert :)

**Comment by soerendomroes** [comment]
@xeruf what is the status here? Should we meet and merge this?

Have you already tested whether the gui works with this?

File: settings.gradle.kts [hunk 3]
```diff
@@ -3,5 +3,13 @@ rootProject.buildFileName = "gradle/build.gradle.kts"
 
 includeBuild("gradle/custom-tasks")
 
-include("sdk", "server", "plugin", "plugin2025", "plugin2026", "player", "test-client")
+// TODO rename to name of game rather than plugin from 2027
```
Comment (soerendomroes) [inline settings.gradle.kts:6]: What is the plan here? 

This might be a good place to document how a new plugin should be added.

I guess that the name plugin2023 and so on must be used in the gui. In the gui build.gradle should hence be some documentation on how to choose a specific plugin.

As I understand this, plugin202X is only called like this, since it is the name used in the gui. Moreover, I remember that the gui just uses the highest number, e.g. plugin2099 would be used if it exists. Is this correct?

Line 12 and onwards maps these plugins to actual projects.
Hence, to add a new project, I create a folder with everything inside call it <newgame>, add plugin20XX and everything is ready?
